crates/web-client/src/models/account_id.rs

```diff
@@ -40,10 +42,31 @@ impl AccountId {
         self.0.to_string()
     }
 
+    /// Will turn the Account ID into its bech32 string representation.
+    /// To avoid a potential wrongful encoding, this function will
+    /// expect only IDs for either mainnet ("mm"), testnet ("mtst") or devnet ("mdev").
+    /// To use a custom bech32 prefix, use `Self::to_bech_32_custom`.
     #[wasm_bindgen(js_name = "toBech32")]
-    pub fn to_bech32(&self) -> String {
-        let network_id = NetworkId::Testnet;
-        self.0.to_bech32(network_id)
+    pub fn to_bech32(&self, network_id: &str) -> Result<String, String> {
+        match NetworkId::from_str(network_id) {
+            Ok(NetworkId::Custom(_)) => {
+                Err("Expected network id for either mainnet, testnet or devnet".to_owned())
```

Comment:
nit:
Miden repos usually start [error messages in lowercase ](https://rust-lang.github.io/api-guidelines/interoperability.html#error-types-are-meaningful-and-well-behaved-c-good-err:~:text=The%20error%20message%20given%20by%20the%20Display%20representation%20of%20an%20error%20type%20should%20be%20lowercase%20without%20trailing%20punctuation%2C%20and%20typically%20concise.)(and also, messages should omit punctuation)
```suggestion
                Err("expected network id for either mainnet, testnet or devnet".to_owned())
```

Reply:
Fixed [here](https://github.com/0xMiden/miden-client/pull/1043/commits/506d925140bed65ebdd51485f33521ea0f9590bf)